CONTRIBUTING.md

```diff
@@ -1,3 +1,542 @@
+
+This is so common there is a multitude of tools to help with this, and the process is now officially supported within python3 itself.
+
+**Bonus Points**
+Watch: Reverse-engineering Ian Bicking's brain: inside pip and virtualenv
+https://www.youtube.com/watch?v=DlTasnqVldc
+
+##### What about Python Package Management
+
+Okay so virtualenvs are only part of the process, they give you isolated folder structures in which you can install, update and delete packages without worrying about messing up other projects. But how do I install a package? Is that only pip, what about conda or pipenv or poetry?
+
+Most of these tools aim to provide the same functionality which is to create virtualenvs, and handle the installation of packages as well as making the experience of activating and managing virtualenvs as seamless as possible. Some, as in the case of conda even provide their own package repositories and additional non-python package support.
+
+For the example below I will be using `pipenv` purely because it is extremely simple to use, and is itself simply a pip package which means as long as you have any version of python3 on your system you can use this to bootstrap everything else.
+
+| name       | packages | virtualenvs |
+| ---------- | -------- | ----------- |
+| pip + venv | ✅       | ✅          |
+| pipenv     | ✅       | ✅          |
+| conda      | ✅       | ✅          |
+| poetry     | ✅       | ✅          |
+
+## Pipenv
+
+As you will be running pipenv to create virtualenvs you will want to install pipenv into your normal system python site-packages.
+This can be achieved by simply `pip` installing it from your shell.
+
+```
+$ pip install pipenv
+```
+
+### Common Issues
+
+- what is the difference between pip and pip3?
+  pip3 was introduced as an alias to use the pip package manager from python3 on systems where python 2.x is still used by the operating system.
+  When in doubt use pip3 or check the path and version that your python or pip binary is using.
+- I don't have pip?
+  On some systems like Ubuntu, you need to install pip first with `apt-get install python3-pip` or you can use the new official way to install pip from python:
+
+```
+$ python3 -m ensurepip
+```
+
+## Git Repo
+
+### Forking SyMPC
+
+As you will be making contributions you will need somewhere to push your code. The way you do this is by forking the repository so that your own GitHub user profile has a copy of the source code.
+
+Navigate to the page and click the fork button:
+https://github.com/OpenMined/SyMPC
+
+You will now have a URL like this with your copy:
+https://github.com/\<your-username>/SyMPC
```

Comment:
extra \ here.

Reply:
I am not sure it is extra. If I remove it, the Markdown visualizer of visual code will show `https://github.com//SyMPC` instead of `https://github.com/<your-username>/SyMPC`